MarshalJSON panics if response is empty

`MarshalJSON` methods need to return JSON encodable bytes if there is an empty response or the JSON encoder errors out.

```
--- FAIL: TestRawQueryGroupBy (17.83s)
    raw_query_groupby_test.go:46:
                Error Trace:    raw_query_groupby_test.go:46
                Error:          Received unexpected error:
                                received '500' status code: {"state":"Error","error":"json: error calling MarshalJSON for type *query.GroupedResults: unexpected end of JSON input"}
                Test:           TestRawQueryGroupBy
```

--- a/query/grouped_result.go
+++ b/query/grouped_result.go
@@ -35,6 +35,10 @@ const (
 	defaultTrimTriggerSizeMultiplier = 4
 )
 
+var (
+	emptyJSONResponse = []byte("{}")
+)
+
 // GroupedResults is a collection of result groups.
 type GroupedResults struct {
 	// GroupBy contains a list of field paths to group results by.
@@ -175,7 +179,7 @@ func (r *GroupedResults) TrimIfNeeded() {
 // MarshalJSON marshals the grouped results as a JSON object.
 func (r *GroupedResults) MarshalJSON() ([]byte, error) {
 	if r.IsEmpty() {
-		return nil, nil
+		return emptyJSONResponse, nil
 	}
 	var (
 		limit        = r.Limit

